profiler/src/main/java/com/splunk/opentelemetry/profiler/AgentInternalsFilter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.profiler;
+
+import java.util.Arrays;
+import java.util.function.Predicate;
+
+public class AgentInternalsFilter implements Predicate<String> {
+
+  private static final String[] UNWANTED_PREFIXES =
+      new String[] {
+        "\"Batched Logs Exporter\"",
+        "\"BatchSpanProcessor_WorkerThread-",
+        "\"JFR Recorder Thread\"",
+        "\"JFR Periodic Tasks\"",
+        "\"JFR Recording Scheduler\"",
+        "\"JFR Recording Sequencer\""
+      };
+
+  @Override
+  public boolean test(String stack) {
+    if (Arrays.stream(UNWANTED_PREFIXES).anyMatch(stack::startsWith)) {
```

Comment:
I am concerned about the performance of that as compared to common `for` loop.

Reply:
Nit: you can use a `Trie` here 😄

Reply:
Is this worth the effort? If I understand correctly this test is applied once a second to the trace form each thread which doesn't seem like a lot.

Reply:
Prefer to do the simple thing first and optimize/microbenchmark later if deemed important.